### ISSUE TITLE
Pull not working via Keymetrics.io

### ISSUE THREAD

**Issue by schme16** [Unitech/keymetrics.io#75]
I'm trying to update my app to the latest commit, but when I click on the `pull` button in my https://keymetrics.io dashboard it seems to be doing something, but doesn't actually update the app; when I run the command pm2 pull <appname> from ssh, however, it updates just fine. 

Before: ![image](https://cloud.githubusercontent.com/assets/116081/7580811/d202f9de-f8be-11e4-83a0-92f96d4be72f.png)
After: 
![image](https://cloud.githubusercontent.com/assets/116081/7580837/23542c7c-f8bf-11e4-8319-9b0ca4fbcd11.png)

That feedback area never seems to go away, not sure if it's meant to or not, but here it is after 20mins of waiting: 
![image](https://cloud.githubusercontent.com/assets/116081/7580855/5f132178-f8bf-11e4-9ac2-0a1ab0ca468c.png)

Any ideas?

**Comment by jshkurti** [comment]
Are you using PM2 v.0.12.14 ?
I think it has been fixed.

**Comment by schme16** [comment]
Sure am, but just to be safe I've uninstalled it and installed via npm
install pm2@latest -g, no luck

**Comment by schme16** [comment]
I thought it might be an issue with my using nvm for node, but even after removing nvm and installing 0.12 I get no change whatsoever

**Comment by jshkurti** [comment]
Does Restart button work ?
If Restart button works but not Pull it means reverse interaction from Keymetrics is ok but you might have some git issues.
Perhaps git is asking for a password input or something alike.
I noticed your repo is cloned in http mode. Can you git clone your repo in ssh mode and try again ?

**Comment by schme16** [comment]
I actually switched from ssh to https trying to see if that was the issue haha, but sure thing I'll try again!

I assumed that restart was working, but will actually confirm that and report back

**Comment by schme16** [comment]
OK, Everything seems to work now:
I checked the ports and they were definitely open, so I checked if the restart button was working, it wasn't...
So  next I checked my firewall settings again and it was all perfect, no leads there.
I switched to ssh (no that that should change anything to do with the restart button), to no change.
I tried linking and unlinking the app and server, no luck...

In the end removed all traces of nvm and all versions of node and npm that it had installed and installed 0.12 and the latest npm; reinstalled pm2 and re-linked the server and app, STILL no luck

What seemed to be the issue was that node was being installed only to the root user context, so everytime pm2 was trying to perform and action it was met with `-bash: node: command not found`. I uninstalled node, npm and pm2 again and reinstalled them all to /usr/bin and /usr/local and after a reboot and yet another relinking of the server and app, everything seems to work just as expected.

@jshkurt:i Thanks for all the advice so far!
